esphome/components/daly_bms/daly_bms.cpp

```diff
@@ -0,0 +1,186 @@
+                this->voltage_sensor_->publish_state(((float) ((data[4] << 8) | data[5]) / 10));
+              }
+              if (this->current_sensor_) {
+                this->current_sensor_->publish_state(((float) (((data[8] << 8) | data[9]) - 30000) / 10));
+              }
+              if (this->battery_level_sensor_) {
+                this->battery_level_sensor_->publish_state(((float) ((data[10] << 8) | data[11]) / 10));
+              }
+              break;
+
+            case DALY_REQUEST_MIN_MAX_VOLTAGE:
+              if (this->max_cell_voltage_) {
+                this->max_cell_voltage_->publish_state((float) ((data[4] << 8) | data[5]) / 1000);
+              }
+              if (this->max_cell_volatge_number_) {
+                this->max_cell_volatge_number_->publish_state(data[6]);
+              }
+              if (this->min_cell_voltage_) {
+                this->min_cell_voltage_->publish_state((float) ((data[7] << 8) | data[8]) / 1000);
+              }
+              if (this->min_cell_voltage_number_) {
+                this->min_cell_voltage_number_->publish_state(data[9]);
+              }
+              break;
+
+            case DALY_REQUEST_MIN_MAX_TEMPERATURE:
+              if (this->max_temperature_) {
+                this->max_temperature_->publish_state(data[4] - DALY_TEMPERATURE_OFFSET);
+              }
+              if (this->max_temperature_probe_number_) {
+                this->max_temperature_probe_number_->publish_state(data[5]);
+              }
+              if (this->min_temperature_) {
+                this->min_temperature_->publish_state(data[6] - DALY_TEMPERATURE_OFFSET);
+              }
+              if (this->min_temperature_probe_number_) {
+                this->min_temperature_probe_number_->publish_state(data[7]);
+              }
+              break;
+
+            case DALY_REQUEST_MOS:
+              if (this->status_text_sensor_ != nullptr) {
+                switch (data[4]) {
+                  case 0:
+                    this->status_text_sensor_->publish_state("Stationary");
+                    break;
+                  case 1:
+                    this->status_text_sensor_->publish_state("Charging");
+                    break;
+                  case 2:
+                    this->status_text_sensor_->publish_state("Discharging");
+                    break;
+                  default:
+                    break;
+                }
+              }
+              if (this->charging_mos_enabled_) {
+                this->charging_mos_enabled_->publish_state(data[5]);
+              }
+              if (this->discharging_mos_enabled_) {
+                this->discharging_mos_enabled_->publish_state(data[6]);
+              }
+              if (this->remaining_capacity_) {
+                this->remaining_capacity_->publish_state(
+                    (float) ((data[8] << 24) | (data[9] << 16) | (data[10] << 8) | data[11]) / 1000);
```

Comment:
There is also a helper for `encode_uint16` that could be used for the values using 2 bytes if you want to change.
```suggestion
                    (float) encode_uint32(data[8], data[9], data[10], data[11]) / 1000);
```

Reply:
Thank you for all suggestions